### PR TITLE
feat(bindings): Pass Metadata in grpc binding invocation

### PR DESCRIPTION
**Pull request by DeepanshuA**
Signed-off-by: Deepanshu Agarwal <deepanshu.agarwal1984@gmail.com>

# Description

Adds metadata capability to output binding invocation.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #339 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by DeepanshuA** [comment]
E2E test is failing here, since we are using Dapr 1.8.0, the latest released major version (https://github.com/dapr/js-sdk/blob/c3f6347f632a508944f7b8296298f665f32a9d6f/.github/workflows/test-e2e.yml#L39)
But, it needs change from https://github.com/dapr/dapr/pull/5049, specifically https://github.com/dapr/dapr/blob/490ef74dbd06e13a4557a249afd58d1abecb01af/pkg/grpc/api.go#L547

**Comment by DeepanshuA** [comment]
Commented out E2E test for now - E2E tests should pass now.
Would un-comment after dapr 1.9 release.

File: test/components/binding-redis.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: dapr.io/v1alpha1
```
Comment (shubham1172) [inline test/components/binding-redis.yaml:1]: Can't we reuse the existing MQTT binding?

**Comment by shubham1172** [comment]
> E2E test is failing here, since we are using Dapr 1.8.0, the latest released major version (
> 
> https://github.com/dapr/js-sdk/blob/c3f6347f632a508944f7b8296298f665f32a9d6f/.github/workflows/test-e2e.yml#L39
> 
> )
> But, it needs change from [dapr/dapr#5049](https://github.com/dapr/dapr/pull/5049), specifically https://github.com/dapr/dapr/blob/490ef74dbd06e13a4557a249afd58d1abecb01af/pkg/grpc/api.go#L547

@DeepanshuA is it possible to use some other metadata for testing? i.e. if you consider MQTT, we can override the topic via metadata ([docs](https://docs.dapr.io/reference/components-reference/supported-bindings/mqtt/#set-topic-per-request)) and test it.

File: test/components/binding-redis.yaml (hunk 1 above)
Comment (XavierGeerinck) [reply test/components/binding-redis.yaml:1]: Let's add it for Redis specifically as well maybe as it's a common component?

Comment (shubham1172) [reply test/components/binding-redis.yaml:1]: Sure, shall we remove MQTT then too? It will ease setup. Redis is more commonly available.

Comment (XavierGeerinck) [reply test/components/binding-redis.yaml:1]: I would keep MQTT as we want to test on different platforms, no? MQTT is used quite broadly btw :)

Comment (DeepanshuA) [reply test/components/binding-redis.yaml:1]: To me it seems fine to have redis and mqtt both.
I used redis here specifically, as for redis metadata "key" is crucial (https://github.com/dapr/components-contrib/blob/b18e73d028e1b6553973efd449085a88d0bb9690/bindings/redis/redis.go#L73) or else this output binding won't serve any purpose.

Comment (shubham1172) [reply test/components/binding-redis.yaml:1]: Ok, I would suggest having one working e2e test (from any component for what it's worth), and then uncomment the other one later.

**Review by shubham1172 — DISMISSED** [review]

**Review by shubham1172 — APPROVED** [review]

**Pull request merged**